Topics/Tech_Stacks/GDB_Debug.md

```diff
@@ -0,0 +1,113 @@
+# Introduction how to debug by GDB
+
+## Table of contents
+
+#### What is GDB
+#### What can do by GDB
+#### How to use GDB
+
+
+## What is GDB?
+GDB (GNU Debugger) is a powerful open-source debugger commonly used in the field of computer science and software development, especially in the context of programming and debugging. 
+
+It is a command-line tool that allows developers to inspect and debug programs written in various programming languages, including C, C++, and even some other languages.
+
+Source: [GDB: The GNU Project Debugger](https://www.sourceware.org/gdb/)
+
+
+## What can do by GDB?
```

Comment:
I think we can assume a CSC301 student know about a debugger, so I feel like this is a bit wasteful. Maybe consider to only highlights the differences between GDB and other debuggers?

Reply:
Since Gdb is different with other debug tool, it is based on unix system. Therefore, I need to introduce what can do by GDB.